pkg/api/customization/cluster/validator.go

```diff
@@ -270,3 +275,72 @@ func (v *Validator) accessTemplate(request *types.APIContext, spec *mgmtclient.C
 
 	return nil
 }
+
+// validateGenericEngineConfig allows for additional validation of clusters that depend on Kontainer Engine or Rancher Machine driver
+func (v *Validator) validateGenericEngineConfig(request *types.APIContext, spec *v3.ClusterSpec) error {
+
+	if request.Method == http.MethodPost {
+		return nil
+	}
+
+	clusterName := request.ID
+	prevCluster, err := v.ClusterLister.Get("", clusterName)
```

Comment:
This should move after the check for the `AmazonElasticContainerServiceConfig`. No reason to error here if this is the wrong driver anyways.

Reply:
moved